lib/spack/spack/version.py

```diff
@@ -456,6 +492,45 @@ def intersection(self, other):
 
 class GitVersion(VersionBase):
     """Class to represent versions interpreted from git refs.
+    
+    If the ref is a commit hash then there are two paths of execution:
+    
+    1) a raw comit hash needs a commit lookup to pair it to a spack version
+    2) a commit hash paired with a valid spack version  i.e. [hash]=main
+
+    Git commit versions that are not paried with a known version
```

Comment:
Throughout this paragraph, I think we want "git version" instead of "git commit version"